flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamOverAggregate.scala

```diff
@@ -0,0 +1,199 @@
+    rowRelDataType: RelDataType,
+    inputType: RelDataType)
+  extends SingleRel(cluster, traitSet, inputNode)
+  with OverAggregate
+  with DataStreamRel {
+
+  override def deriveRowType(): RelDataType = rowRelDataType
+
+  override def copy(traitSet: RelTraitSet, inputs: JList[RelNode]): RelNode = {
+    new DataStreamOverAggregate(
+      logicWindow,
+      cluster,
+      traitSet,
+      inputs.get(0),
+      getRowType,
+      inputType)
+  }
+
+  override def toString: String = {
+    s"OverAggregate(${aggOpName})"
+  }
+
+  override def explainTerms(pw: RelWriter): RelWriter = {
+    val (
+      overWindow: Group,
+      partition: Array[Int],
+      namedAggregates: IndexedSeq[CalcitePair[AggregateCall, String]]
+      ) = genPartitionKeysAndNamedAggregates
+
+    super.explainTerms(pw)
+      .itemIf("partitionBy", partitionToString(inputType, partition), partition.nonEmpty)
+        .item("orderBy",orderingToString(inputType, overWindow.orderKeys.getFieldCollations))
+      .item("range", windowRange(overWindow))
+      .item(
+        "select", aggregationToString(
+          inputType,
+          getRowType,
+          namedAggregates))
+  }
+
+  override def translateToPlan(tableEnv: StreamTableEnvironment): DataStream[Row] = {
+    if (logicWindow.groups.size > 1) {
+      throw new TableException(
+        "Unsupported use of OVER windows. All aggregates must be computed on the same window.")
+    }
+
+    val overWindow: org.apache.calcite.rel.core.Window.Group = logicWindow.groups.get(0)
+
+    val inputDS = input.asInstanceOf[DataStreamRel].translateToPlan(tableEnv)
+
+    if (overWindow.orderKeys.getFieldCollations.size() != 1) {
+      throw new TableException(
+        "Unsupported use of OVER windows. The window may only be ordered by a single time column.")
+    }
+
+    val timeType = inputType
+      .getFieldList
+      .get(overWindow.orderKeys.getFieldCollations.get(0).getFieldIndex)
+      .getValue
+
+    timeType match {
+      case _: ProcTimeType =>
+        // both ROWS and RANGE clause with UNBOUNDED PRECEDING and CURRENT ROW condition.
+        if (overWindow.lowerBound.isUnbounded &&
+          overWindow.upperBound.isCurrentRow) {
+          createUnboundedAndCurrentRowProcessingTimeOverWindow(inputDS)
+        } else {
+          throw new TableException(
+              "OVER window only support ProcessingTime UNBOUNDED PRECEDING and CURRENT ROW " +
+              "condition.")
+        }
+      case _: RowTimeType =>
+        throw new TableException("OVER Window of the EventTime type is not currently supported.")
+      case _ =>
+        throw new TableException(s"Unsupported time type {$timeType}")
+    }
+
+  }
+
+  def createUnboundedAndCurrentRowProcessingTimeOverWindow(
+    inputDS: DataStream[Row]): DataStream[Row]  = {
+
+    val (_,
+      partition: Array[Int],
+      namedAggregates: IndexedSeq[CalcitePair[AggregateCall, String]]
+      ) = genPartitionKeysAndNamedAggregates
```

Comment:
change `genPartitionKeysAndNamedAggregates()` to only generate named aggregates and rename.
Functions that do only one thing are easier to understand and partition keys can be generated in one line: `overWindow.keys.toArray`.